universal-application-tool-0.0.1/test/forms/QuestionFormTest.java

```diff
@@ -0,0 +1,52 @@
+package forms;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.OptionalLong;
+import org.junit.Test;
+import services.question.QuestionDefinition;
+import services.question.QuestionDefinitionBuilder;
+import services.question.TextQuestionDefinition;
+import services.question.UnsupportedQuestionTypeException;
+
+public class QuestionFormTest {
+
+  @Test
+  public void getBuilder_returnsCompleteBuilder() throws UnsupportedQuestionTypeException {
+    QuestionForm form = new QuestionForm();
+    form.setQuestionName("name");
+    form.setQuestionDescription("description");
+    form.setQuestionPath("my.question.path");
+    form.setQuestionText("What is the question text?");
+    form.setQuestionHelpText("");
+    form.setQuestionType("TEXT");
+    QuestionDefinitionBuilder builder = form.getBuilder();
+
+    // The QuestionForm does not set version or id, which are needed in order to build the
+    // QuestionDefinition.
+    builder.setVersion(1L);
+
+    TextQuestionDefinition expected =
+        new TextQuestionDefinition(
+            OptionalLong.empty(),
```

Comment:
nit (blocking): we can omit specifying this now. Please use the constructor that doesn't take an `id` as argument.

Reply:
Done.